mogp_emulator/HistoryMatching.py

```diff
@@ -0,0 +1,650 @@
+        :type expectations: tuple of 3 ndarrays or None
+        :returns: None
+        """
+        # need to allow expectations == None, as otherwise can't reset values
+        if not self.check_expectations(expectations) and (not expectations is None):
+            raise TypeError("bad input for set_expectations - expected a Tuple " +
+                            "of 3 ndarrays.")
+        self.expectations = expectations
+        self.update()
+
+
+    def set_threshold(self, threshold):
+        r"""
+        Set the threshold value for history matching
+
+        Sets the ``self.threshold variable`` to the provided threshold argument
+        if it  passes the ``check_threshold`` requirements. The threshold must
+        be a non-negative float.
+
+        :param threshold: New value for threshold, must be a non-negative float.
+        :type threshold: float
+        :returns: None
+        """
+        if not self.check_threshold(threshold):
+            raise TypeError("bad input for set_expectations - expected a float")
+        self.threshold = float(threshold)
+
+
+    def status(self):
+        r"""
+        Prints a summary of the current status of the class object.
+
+        Prints a summary of the current status of the class object, including the
+        values stored in ``self.gp``, ``.obs``, ``.ndim``, ``.ncoords``, and
+        ``.threshold``, and the shapes/sizes ofvalues stored in ``self.obs``,
+        ``.coords``, ``.expectations``, ``.I``, ``.NROY``, and ``.RO.``
+        No inputs, no return value.
+
+        :returns: None
+        """
+
+        print(str(self))
+
+
+    def check_gp(self, gp):
+        r"""
+        Checks if the provided argument is consistent with expectations for a GP.
+
+        Returns a boolean that is True if the provided quantity is consistent with
+        the requirements for a gaussian process, i.e. is of type ``GaussianProcess``.
+
+        :param gp: Input GP object to be checked.
+        :type gp: GaussianProcess
+        :returns: Boolean indicating if provided object is a ``GaussianProcess``
+        :rtype: bool
+        """
+        if gp is None: return False
+        if isinstance(gp, GaussianProcess):
+            return True
+        return False
+
+
+    def check_obs(self, obs):
+        r"""
+        Checks if the provided argument is consistent with expectations for
+        observations.
+
+        Returns a boolean that is ``True`` if the provided quantity is consistent with
+        the requirements for the observation quantity, i.e. is a numerical value
+        that can be converted to a float, or a list of up to two such values. Also
+        checks if the provided variance is non-negative.
+
+        :param obs: Input for observations to be checked
+        :type obs: float or list
+        :returns: Boolean indicating if provided observations are acceptable
+        :rtype: bool
+        """
+        # TODO: allow tuple arguments as well as lists.

```

Comment:
TODO comment

Reply:
Fixed this, now allows tuples and numpy arrays as well.